sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -216,7 +216,7 @@ class PlannerSuite extends SharedSQLContext {
           ).queryExecution.executedPlan.collect {
             case exchange: ShuffleExchange => exchange
           }.length
-          assert(numExchanges === 5)
+          assert(numExchanges === 3)
```

Comment:
Do you apply all changes? I can't reproduce it and Jenkins tests are passed too.